tycho-sbom/src/main/java/org/eclipse/tycho/sbom/TychoProjectDependenciesConverter.java

```diff
@@ -117,35 +120,37 @@ private void convertToDependency(DependencyTreeNode node, Set<Dependency> depend
 	private List<String> getBomRepresentation(IInstallableUnit iu) {
 		final MavenSession mavenSession = legacySupport.getSession();
 		final List<MavenProject> reactorProjects = mavenSession.getAllProjects();
-		// mutable!
-		List<String> bomRefs = new ArrayList<>();
-		// (I) IU describes local reactor project
-		try {
-			ProjectDependencyClosure dependencyClosure = dependencyProcessor
-					.computeProjectDependencyClosure(reactorProjects, mavenSession);
-			MavenProject iuProject = dependencyClosure.getProject(iu).orElse(null);
-			if (iuProject != null) {
-				String bomRef = modelConverter.generatePackageUrl(iuProject.getArtifact());
-				if (bomRef == null) {
-					LOG.error("Unable to calculate BOM for: " + iuProject);
+		return bomRepresentations.computeIfAbsent(iu, ignore -> {
+			// mutable!
+			List<String> bomRefs = new ArrayList<>();
+			// (I) IU describes local reactor project
+			try {
+				ProjectDependencyClosure dependencyClosure = dependencyProcessor
+						.computeProjectDependencyClosure(reactorProjects, mavenSession);
+				MavenProject iuProject = dependencyClosure.getProject(iu).orElse(null);
+				if (iuProject != null) {
+					String bomRef = modelConverter.generatePackageUrl(iuProject.getArtifact());
+					if (bomRef == null) {
+						LOG.error("Unable to calculate BOM for: " + iuProject);
+						return bomRefs;
+					}
+					bomRefs.add(bomRef);
 					return bomRefs;
 				}
-				bomRefs.add(bomRef);
-				return bomRefs;
+			} catch (CoreException e) {
+				LOG.error(e.getMessage(), e);
+				return Collections.emptyList();
 			}
-		} catch (CoreException e) {
-			LOG.error(e.getMessage(), e);
-			return Collections.emptyList();
-		}
-		// (II) IU describes external artifact
-		for (IArtifactKey p2artifactKey : iu.getArtifacts()) {
-			String bomRef = modelConverter.generateP2PackageUrl(p2artifactKey, true, true, false);
-			if (bomRef == null) {
-				LOG.error("Unable to calculate BOM for: " + p2artifactKey);
-				continue;
+			// (II) IU describes external artifact
+			for (IArtifactKey p2artifactKey : iu.getArtifacts()) {
+				String bomRef = modelConverter.generateP2PackageUrl(p2artifactKey, true, true, false);
+				if (bomRef == null) {
+					LOG.error("Unable to calculate BOM for: " + p2artifactKey);
+					continue;
+				}
+				bomRefs.add(bomRef);
 			}
-			bomRefs.add(bomRef);
-		}
-		return bomRefs; // mutable!
+			return bomRefs; // mutable!
```

Comment:
The APi says the returned list is mutable, so either drop the mutability (and use a read-only list) or return a copy of the cached value.

Reply:
It's been a few months, but I believe mutability was required in an earlier implementation because CycloneDX would add element to this list. But now it's only used within loops, so I think dropping this property is the best way to go.